File: backend/src/main/kotlin/com/skewwhiffy/notation/model/abc/ClefLineNotes.kt
package com.skewwhiffy.notation.model.abc

import com.skewwhiffy.notation.model.note.AbsoluteNote
import com.skewwhiffy.notation.model.clef.Clef

data class ClefLineNotes(private val clef: Clef) : ClefNotes {
  override val notes: List<AbsoluteNote> = (1..4)
    .fold(listOf(clef.lowLedgerNote.skipOne)) { soFar, _ -> soFar + soFar.last().skipOne }
    .map { it.withNoteName }
}

File: backend.scala/app/com/skewwhiffy/auraltester/clefs/ClefLineNotes.scala
package com.skewwhiffy.auraltester.clefs

import com.skewwhiffy.auraltester.notes.AbsoluteNote

case class ClefLineNotes(clef: Clef) extends ClefNotes {
  def notes: List[AbsoluteNote] = {
    val lowestNote = clef.lowLedgerNote.skipOne
    Range(1, 5)
      .foldLeft(List(lowestNote))((soFar, _) => soFar :+ soFar.last.skipOne)
      .map(it => it.withNoteName)
  }
}
